### PR TITLE
[Mellanox]Revert "Skip available route check for mellanox platforms (#12787)"

### PR DESCRIPTION
**Pull request by JibinBao**
### Description of PR

Revert https://github.com/sonic-net/sonic-mgmt/pull/12787, because there is another fix: https://github.com/sonic-net/sonic-mgmt/pull/12882#

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [x] 202305
- [x] 202311
- [x] 202405

#### What is the motivation for this PR?
Revert the PR, so the  crm test can run on mellanox device

#### How did you do it?
Revert the PR

#### How did you verify/test it?
run the test_crm_route on mellanox device 

#### Any platform specific information?
any

### CODE REVIEW

**Review by prabhataravind — APPROVED** [review]
LGTM, please merge this after merging https://github.com/sonic-net/sonic-mgmt/pull/12882

**Comment by prabhataravind** [comment]
@bingwang-ms for viz

**Review by bingwang-ms — APPROVED** [review]

**Pull request merged**
